tests/iso20/ac_charge_loop.cpp

```diff
@@ -0,0 +1,167 @@
+#include <catch2/catch_test_macros.hpp>
+
+#include <array>
+#include <vector>
+
+#include <cbv2g/iso_20/iso20_AC_Decoder.h>
+#include <cbv2g/iso_20/iso20_AC_Encoder.h>
+
+#include "test_utils/codec.hpp"
+
+// Exi Stream: 8008041e9869d6a61dc1ef895b9b4a8062832418640096
+// XML:
+// <AC_ChargeLoopReq>
+//     <Header>
+//         <SessionID>3D30D3AD4C3B83DF</SessionID>
+//         <TimeStamp>1695358101</TimeStamp>
+//     </Header>
+//     <MeterInfoRequested>false</MeterInfoRequested>
+//     <BPT_Scheduled_AC_CLReqControlMode>
+//         <EVPresentActivePower>
+//             <Exponent>3</Exponent>
+//             <Value>200</Value>
+//         </EVPresentActivePower>
+//     </BPT_Scheduled_AC_CLReqControlMode>
+// </AC_ChargeLoopReq>
+
+// Exi Stream: 800c041e9869d6a61dc1ef895b9b4a8062005900
+// XML:
+// <AC_ChargeLoopRes>
+//     <Header>
+//         <SessionID>3D30D3AD4C3B83DF</SessionID>
+//         <TimeStamp>1695358101</TimeStamp>
+//     </Header>
+//     <ResponseCode>OK</ResponseCode>
+//     <BPT_Scheduled_AC_CLResControlMode/>
+// </AC_ChargeLoopRes>
+
+static void setup_header(struct iso20_ac_MessageHeaderType* header,
+                         const std::array<uint8_t, iso20_ac_sessionIDType_BYTES_SIZE>& session_id, uint64_t timestamp) {
+    init_iso20_ac_MessageHeaderType(header);
+    header->SessionID.bytesLen = iso20_ac_sessionIDType_BYTES_SIZE;
+    std::copy(session_id.begin(), session_id.end(), header->SessionID.bytes);
+    header->TimeStamp = timestamp;
+}
+
+SCENARIO("Encode and decode AC charge loop message") {
```

Comment:
You might want to add ISO 15118-20 or "ISO 20" to the test case names.

Reply:
Done